# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/GtkCompositorWidget.cpp
# Commit: 69e7c8a0cca3
# Full Hash: 69e7c8a0cca3930210bb22918b5d0f808372ee48
# Author: stransky <stransky@redhat.com>
# Date: 2024-07-07 20:51:50
# Regressor Bug: 1898476
# File Overlap Count: 1
# Description:
#   Bug 1898476 [Wayland] Provide surface lock by GtkCompositorWidget r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D214884
# ==============================================================================

diff -r 480f3f701046 -r 69e7c8a0cca3 widget/gtk/GtkCompositorWidget.cpp
--- a/widget/gtk/GtkCompositorWidget.cpp	Sun Jul 07 08:46:39 2024 +0000
+++ b/widget/gtk/GtkCompositorWidget.cpp	Sun Jul 07 08:46:40 2024 +0000
@@ -211,5 +211,11 @@
 }
 #endif
 
+#if defined(MOZ_WAYLAND)
+UniquePtr<MozContainerSurfaceLock> GtkCompositorWidget::LockSurface() {
+  return mWidget->LockSurface();
+}
+#endif
+
 }  // namespace widget
 }  // namespace mozilla